tensorflow/compiler/xla/service/slice_delaying.cc

```diff
@@ -0,0 +1,306 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/xla/service/slice_delaying.h"
+#include <algorithm>
+#include <utility>
+
+namespace xla {
+
+bool SliceLess(const HloInstruction* lhs, const HloInstruction* rhs) {
+  int64 l = 0;
+  int64 r = 0;
+  for (int i = 0; i < lhs->slice_starts().size(); ++i) {
+    l = lhs->slice_starts(i);
+    r = rhs->slice_starts(i);
+    if (l < r)
+      return true;
+    if (l > r)
+      return false;
+  }
+  return false;
```

Comment:
Is this function simply

```
return lhs->slice_starts() < rhs->slice_starts();
```

?  In which case we can just inline it into the std::sort call below using a lambda?